src/entitysdk/result.py

```diff
@@ -0,0 +1,48 @@
+"""Iterator wrapper for iterable results."""
+
+from collections.abc import Iterable, Iterator
+from typing import Self, TypeVar
+
+from entitysdk.exception import IteratorResultError
+
+ResultType = TypeVar("ResultType")
+
+
+class IteratorResult(Iterator[ResultType]):
+    """A result of an iterator."""
+
+    def __init__(self, iterable: Iterable[ResultType]) -> None:
+        """Initialize the iterator result."""
+        self._iterable = iter(iterable)
+
+    def __iter__(self) -> Self:
+        """Return the iterator."""
+        return self
+
+    def __next__(self) -> ResultType:
+        """Return the next element of the iterable."""
+        return next(self._iterable)
+
+    def first(self) -> ResultType:
+        """Return the first element of the iterable."""
+        if first_item := next(self, None):
+            return first_item
+        raise IteratorResultError("Iterable is empty.")
```

Comment:
Since we are mimicking the behaviour of `first` from sqlalchemy, we should return the first result, or None if the result doesn’t contain any row, not raise an error.

Reply:
Changed it to behave that way.